tests/shardddl2/run.sh

```diff
@@ -0,0 +1,715 @@
+}
+
+function DM_051_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} change a c int;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} change b c int;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6);"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} change b c int;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9);"
+
+    if [[ "$1" = "pessimistic" ]]; then
+        check_log_contain_with_retry "is different with" $WORK_DIR/master/log/dm-master.log
+    else
+        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "query-status test" \
+            "because schema conflict detected" 2
+    fi
+}
+
+function DM_051() {
+    run_case 051 "double-source-pessimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int);\"" \
+    "clean_table" "pessimistic"
+    run_case 051 "double-source-optimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int);\"" \
+    "clean_table" "optimistic"
+}
+
+function DM_056_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} change a c int after b;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} change a c int first;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6);"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} change a c int first;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9);"
+
+    if [[ "$1" = "pessimistic" ]]; then
+        check_log_contain_with_retry "is different with" $WORK_DIR/master/log/dm-master.log
+    else
+        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "query-status test" \
+            "because schema conflict detected" 2
```

Comment:
seems this result is not consistent with the document, need to update the document

Reply:
Have updated it.